internal/model/plugin.go

```diff
@@ -0,0 +1,49 @@
+package model
+
+import (
+	"encoding/json"
+	"io"
+
+	mattermostModel "github.com/mattermost/mattermost-server/model"
+)
+
+// Plugin represents a Mattermost plugin in the marketplace.
+type Plugin struct {
+	HomepageURL       string
+	DownloadURL       string
+	DownloadSignature []byte
+	Manifest          *mattermostModel.Manifest
```

Comment:
I'm probably to late with my comment, but why is the whole manifest included in the data struct? I'm concern is that we now have two sources of truth: The data struct and the file in the repo. My proposal would be to have a link to the manifest instead.

Reply:
We need the manifest "up front" to enable the marketplace and potentially the server to make informed decisions about a plugin without having to download anything. For example, we have the existing `min_server_version` which would be helpful in driving decisions about what to display. We're also looking at a `required_config` setting.

Reply:
>  potentially the server to make informed decisions about a plugin without having to download anything

Why is this a requirement?

Reply:
At a minimum, the marketplace needs to be able to filter down plugins to only show versions applicable to the requesting server in question. It won't be able to do that without the corresponding bit from the manifest. We of course need the id, name, description too. We might want the new `required_config` bit at some point. 

At this point, it seems simpler to just include the whole manifest than conditionally throw things away. Thoughts?

Reply:
My concern was that the JSON DB would get out of sync with the repo. But given that we update the DB after every release of a plugin (right?), this should not be a problem.

I thing this is fine for now. Maybe I'm thinking a bit to much about how this will play out in the future.